### DISCUSSION TITLE
Associate two tables

### DISCUSSION THREAD

**Discussion by 467481** [planetscale/discussion#340] [Q&A] [answered]
Hi, I am a new developer and for a personal project I am trying to create a springboot API for a ReactJS frontend. I switched from a docker mysql container to planetscale. However, I am having trouble with the no foreign key thing. In my database, I have a table User with fields: id, username, password. Other table is user_role with fields: id, user_id, role_name. With my previous mysql container I used foreign keys to create a relation between the two tables. How would I be able to create the relation with planetscale?

**Comment by fooforge** [comment, marked as answer]
Hi @467481,

This is a great question as it's a common misconception that foreign keys are not supported. They are, in fact, supported! What is not supported, however, are foreign key **constraints**. This means, you're totally good to use foreign keys to create relations between tables, but referential integrity you will need to control on the application level.

We have that documented fairly thoroughly in [this article](https://planetscale.com/docs/learn/operating-without-foreign-key-constraints) and I would recommend you to read through it. Afterwards, check out Shlomi's elaborate response in https://github.com/planetscale/discussion/discussions/74#discussioncomment-1196261 which explains some more nuances and goes into deletion strategies.
